File: 프로그래머스/Programmers12980.java
import java.util.*;

public class Solution {
    public int solution(int n) {
        int ans = 1;
        if(n%2 ==0) ans =0;
        long point =1;
        while(n > 1){
            n /=2;
            if(n % 2!=0){
                n*=2;
                n++;
                n/=2;
                ans++;
                continue;
            }
        }

        return ans;
    }
}

File: Programmers12980.js
function solution(n) {
    let ans = 0;
    

    if (n % 2 !== 0) {
        ans++;
        n--; 
    }


    while (n > 0) {
        n /= 2; 
        if (n % 2 !== 0) {
            ans++; 
            n--; 
        }
    }
    
    return ans;
}
